client/state/selectors/get-site-default-post-format.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getRawSite, getSiteOption } from 'state/sites/selectors';
+import { getSiteSettings } from 'state/site-settings/selectors';
+
+/**
+ * Returns the default post format of a site.
+ * Returns null if the site is unknown.
+ *
+ * @param  {Object}    state   Global state tree
+ * @param  {Number}    siteId  The ID of the site we're querying
+ * @return {?String}           The default post format of that site
+ */
+export default function getSiteDefaultPostFormat( state, siteId ) {
+	const site = getRawSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+
+	let defaultPostFormat = get( getSiteSettings( state, siteId ), 'default_post_format' );
+	if ( ! defaultPostFormat ) {
+		defaultPostFormat = getSiteOption( state, siteId, 'default_post_format' );
+	}
+
+	if ( ! defaultPostFormat || defaultPostFormat === '0' ) {
```

Comment:
To the above point, two things:

- Settings and site data are separately retrieved from the API, so it's not enough to check just that the raw site data is available
- I'm not sure if we want to assume a "standard" format in the case that we don't know what the site or site settings are; typically we'd return `null` from a selector in those cases to indicate a "can't know" state.